signature/plugin.go

```diff
@@ -259,3 +268,10 @@ func parseCertChain(certChain [][]byte) ([]*x509.Certificate, error) {
 	}
 	return certs, nil
 }
+
+// Equal reports whether d and t points to the same content.
+func equal(d *ocispec.Descriptor, t *ocispec.Descriptor) bool {
+	return d.MediaType == t.MediaType &&
+		d.Digest == t.Digest &&
+		d.Size == t.Size
+}
```

Comment:
Replace with `content.Equal()` from `oras-go`?

Reply:
Updated the code. But note this PR is mainly about the verification refactoring. Please skip the signature package for now.